### PR TITLE
Add docker support in makefile

### PR DESCRIPTION
**Pull request by anaelChardan**
**Description (for Contributor and Core Developer)**

This PR adds the support for docker in makefile

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,19 +1,55 @@
...
+
+.PHONY: help-community
+help-community: ## Show the community help
+	@echo ""
+	@echo "Pim Community Dev available targets: "
+	@echo ""
+	@echo "You can use docker by using an environment variable, PLATFORM=docker"
+	@echo ""
+	@echo "========================================"
+	@awk 'BEGIN {FS = ":.*?## "} /^[a-zA-Z_-]+:.*?## / {printf "\033[36m%-$(HELP_TARGET_COLUMN_WIDTH)s\033[0m %s\n", $$1, $$2}' $(MAKEFILE_LIST) | sort
```
Comment (BitOne) [inline Makefile]: Nop, please don't force color output...You know how it can end when you want to analyze it...

Comment (anaelChardan) [reply Makefile]: Fixed ! ;) I totally agree :D

**Comment by jjanvier** [comment]
This seems to add a lot of complexity :/ 

Can't the people who want to use the Makefile log into their container, and execute the Makefile?
What's the problem with that?

**Comment by anaelChardan** [comment]
@jjanvier the goal is to be transparent for everyone, yes they can but they won't benefit from the autocompletion and so on, it's the matter of 2 variables :D

**Comment by jjanvier** [comment]
When the Makefile was added, I saw it as a big opportunity to simplify our recurring tasks. But when I see that PR, I'm afraid I won't contribute as much or easily as I expected.

**Comment by anaelChardan** [comment]
To add a task it's totally transparent what are you afraid of? Of adding `eval ${PHP_EXEC} ` in front of your task ?

**Comment by BitOne** [comment]
I agree with JJ. With this PR, half of the makefile is to handle something that's not the purpose of the makefile...

**Comment by jjanvier** [comment]
Not, it's not "2 variables". Yes, `eval ${PHP_EXEC} ` is much more complex than `php`. 

I expected we could have something like that https://speakerdeck.com/mykiwi/outils-pour-ameliorer-la-vie-des-developpeurs-symfony. (Or video https://www.youtube.com/watch?v=_OEDoPMvNY4).

Their Makefile is really great, I hope we'll have something like that. But with your solution it will soon be very complex to handle. 

Docker should be transparent right? Isn't it the original promise? Why should we change our code for Docker? It's a non sense imo...

**Comment by anaelChardan** [comment]
@jjanvier In the link you provided the Makefile is simpler as the makefile uses docker directly ;)

File: Makefile [hunk 2]
```diff
@@ -1,19 +1,28 @@
+DOCKER ?= false
+PHP_EXEC_CE :=""
```
Comment (BitOne) [inline Makefile]: PHP_EXEC_CE => PHP_EXEC ;)

File: Makefile [hunk 3]
```diff
@@ -1,19 +1,28 @@
...
+NODE_EXEC_CE :=""
```
Comment (BitOne) [inline Makefile]: Same ;)

**Comment by jjanvier** [comment]
In the link I provided, there are 2 parts:
1. the original makefile, without any Docker
2. the adaptation for Docker by using variables such as $(EXEC_PHP)

In their team, they all use Docker so yes it's more simple. 

Can we get rid of the `eval` at least? OK to write something like `$(EXEC_PHP) the command` if it's the compromise we have to do so that everybody can use the Makefile.

At least your PR is easier to understand now that it was before.

**Comment by anaelChardan** [comment]
@jjanvier you can GTM it is as simple as you wish I hope ;)

**Comment by damien-carcel** [comment]
Makefile cannot be executed in the containers. If, for instance, you want to add a `make front`, you need to call both PHP (for Symfony assets) and Yarn (for webpack), so 2 containers.
IMHO, Makefile has to be transparent, or it will be only for those who don't use Docker.

**Pull request merged**